src/dda/utils/git/changeset.py

```diff
@@ -182,7 +151,37 @@ def from_patches(cls, diff_output: str | list[str]) -> Self:
             # Strip every "block" and add the missing separator
             patch = "" if binary else "\n".join([sep + block.strip() for block in blocks]).strip()
             changes.append(ChangedFile(path=current_file, type=current_type, binary=binary, patch=patch))
-        return cls.from_iter(changes)
+        return cls(changes)
+
+    def __or__(self, other: Self) -> Self:
+        return type(self)(chain(self.files, other.files))
```

Comment:
I couldn't really find a way to satisfy this other than having a `classmethod` (like the removed `from_iter`) that returns an instance of `cls`.